lib/hexdocs/search.ex

```diff
@@ -0,0 +1,80 @@
+defmodule Hexdocs.Search do
+  @moduledoc """
+  Module responsible for indexing the documentation.
+
+  Currently uses [Typesense Cloud.](https://cloud.typesense.org)
+  """
+
+  require Logger
+
+  @collection "hexdocs"
+
+  @spec process_results(String.t(), Version.t(), [map]) :: :ok
+  def index(package, version, search_items) do
+    full_package = "#{package}-#{version}"
+
+    ndjson =
+      Enum.map(search_items, fn item ->
+        json =
+          Map.take(item, ["type", "ref", "title", "doc"])
+          |> Map.put("package", full_package)
+          |> Jason.encode_to_iodata!()
+
+        [json, ?\n]
+      end)
+
+    url = url("collections/#{@collection}/documents/import?action=create")
+    headers = headers([{"content-type", "text/plain"}])
+
+    case :hackney.post(url, headers, ndjson) do
+      {:ok, 200, _resp_headers, ref} ->
+        process_results(package, version, ref)
```

Comment:
Typesense responds with NDJSON of "status" jsons for each indexed document: https://typesense.org/docs/26.0/api/documents.html#index-multiple-documents

```json
{"success": true}
{"success": false, "error": "Bad JSON.", "document": "[bad doc]"}
```